network/src/transport/mod.rs

```diff
@@ -217,16 +221,19 @@ async fn upgrade_inbound<T: TSocket>(
 
     // try authenticating via noise handshake
     let (mut socket, remote_peer_id) = ctxt.noise.upgrade_inbound(socket).await.map_err(|err| {
-        // security logging
-        warn!(
-            SecurityEvent::InvalidNetworkPeer,
-            error = err.to_string(),
-            network_address = addr,
-            "security: InvalidNetworkPeer {}: '{}'",
-            addr,
-            err
-        );
-        err
+        if err.should_security_log() {
+            sample!(
+                SampleRate::Duration(Duration::from_secs(15)),
+                error!(
```

Comment:
In my opinion, an error should be reserved for internal issues that need attention, this definitely feels more like a warn or maybe an info.